blueocean-rest/pom.xml

```diff
@@ -24,4 +24,47 @@
           <artifactId>capability-annotation</artifactId>
       </dependency>
   </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <dependencies>
+                    <dependency>
+                        <groupId>de.skuzzle.enforcer</groupId>
+                        <artifactId>restrict-imports-enforcer-rule</artifactId>
+                        <version>0.7.0</version>
+                    </dependency>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>check-no-jenkins-imports</id>
+                        <phase>process-sources</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <restrictImports implementation="de.skuzzle.enforcer.restrictimports.RestrictImports">
+                                    <basePackage>**</basePackage>
```

Comment:
We could add a second rule to allow certain imports for a new `io.jenkins.blueocean.rest.factory` package. All factory classes would be moved from `blueocean-rest-impl` to `blueocean-rest`. Then plugins could safely depend on `blueocean-rest` and use the factories.